Configurations/unix-Makefile.tmpl

```diff
@@ -942,71 +947,77 @@ EOF
                                     my $f = basename($_);
                                     (my $l = $f) =~ s/^lib//;
                                     " -L$d -l$l" } @{$args{deps}});
-      my $deps = join(" ",compute_lib_depends(@{$args{deps}}));
-      my $shlib_target = $target{shared_target};
-      my $ordinalsfile = defined($args{ordinals}) ? $args{ordinals}->[1] : "";
+      my @objs = map { (my $x = $_) =~ s|\.o$|$objext|; $x }
+                 grep { $_ =~ m|\.o$| }
+                 @{$args{objs}};
+      my @defs = grep { $_ =~ /\.(def|map)/ } @{$args{objs}};
+      my @deps = compute_lib_depends(@{$args{deps}});
+      my $objs = join(" ", @objs);
+      my $deps = join(" ", @defs, @deps);
       my $target = shlib_simple($lib);
       my $target_full = shlib($lib);
-      return <<"EOF"
+      my $shared_soname = "";
+      $shared_soname .= ' '.$target{shared_sonameflag}.basename($target_full)
+          if defined $target{shared_sonameflag};
+      my $shared_imp = "";
+      $shared_imp .= ' '.$target{shared_impflag}.basename($target)
+          if defined $target{shared_impflag};
+      my $shared_def = join("", map { ' '.$target{shared_defflag}.$_ } @defs);
+      my $recipe = <<"EOF";
 # With a build on a Windows POSIX layer (Cygwin or Mingw), we know for a fact
```

Comment:
Since you're in the area, should "With a build on..." be changed to "When building on..." ?  Side-note.

Reply:
Will fix